core/chains/evm/client/node_lifecycle.go

```diff
@@ -258,19 +319,20 @@ func (n *node) outOfSyncLoop(stuckAtBlockNumber int64) {
 				n.declareUnreachable()
 				return
 			}
-			if head.Number > stuckAtBlockNumber {
+			if until.alive(head) {
```

Comment:
I think this check is not quite right, because it could mark a node alive again that is still lagging behind. It would be more complex, but I think ideally we'd check against the latest head here and make sure we were less than the threshold behind latest.